src/vs/workbench/api/common/extHostLanguageFeatures.ts

```diff
@@ -1355,6 +1355,64 @@ class InlineCompletionAdapter extends InlineCompletionAdapterBase {
 	}
 }
 
+class InlineEditAdapter {
+	private languageTriggerKindToVSCodeTriggerKind: Record<languages.InlineEditTriggerKind, InlineEditTriggerKind> = {
+		[languages.InlineEditTriggerKind.Automatic]: InlineEditTriggerKind.Automatic,
+		[languages.InlineEditTriggerKind.Invoke]: InlineEditTriggerKind.Invoke,
+	};
+
+	async provideInlineEdits(uri: URI, context: languages.IInlineEditContext, token: CancellationToken): Promise<languages.IInlineEdit | undefined> {
+		let disposableStore: DisposableStore | undefined = undefined;
```

Comment:
`disposableStore` is leaking! Look at how inline completions deal with this:

```
const pid = this._references.createReferenceId({
	dispose() {
		disposableStore?.dispose();
	},
	items: normalizedResult
});
```
Unfortunately, getting this right adds quite a lot of complexity...